common_patches/usb0.yaml

```diff
@@ -0,0 +1,395 @@
+_array:
+  DIEPTXF*: {}
+_cluster:
+  IN_EP0:
+    description: "Device IN endpoint 0"
+    DIEPCTL0:
+      name: DIEPCTL
+      _modify:
+        D_MPS0:
+          name: MPS
+        D_USBACTEP0:
+          name: USBACTEP
+        D_NAKSTS0:
+          name: NAKSTS
+        D_EPTYPE0:
+          name: EPTYPE
+        D_STALL0:
+          name: STALL
+        D_TXFNUM0:
+          name: TXFNUM
+        D_CNAK0:
+          name: CNAK
+        DI_SNAK0:
+          name: SNAK
+        D_EPDIS0:
+          name: EPDIS
+        D_EPENA0:
+          name: EPENA
+    DIEPINT0:
```

Comment:
Note that the DIEPINT register definition is the same for endpoint 0 as for the other endpoints.  However, with this code svd2rust generates different types for endpoint 0 than the other endpoints, which is still a bit unfortunate.

We might be able to address this by having separate clusters for each endpoint register, so that you access usb0.diepint(N) rather than usb0.in_ep(N - 1).diepint().  However, this seems like it would make the clustering even more confusing if some endpoint clusters start at endpoint 0 and others start at endpoint 1.

Reply:
I think I can fix this in the SVD file by specifying a `derivedFrom` to indicate that the `IN_EP%s.DIEPINT` registers derive from `IN_EP0.DIEPINT`.  I've confirmed that svd2rust generates the desired code when manually modifying the SVD file with the `derivedFrom` attribute.

However, it looks like svdtools currently isn't capable of generating a patch that can do this.  (It doesn't really handle deriving and clustering together.)  I'll look into how difficult it might be to come up with a reasonable way to fix this in svdtools.

In the meantime, I've worked around this in my driver code by just manually doing an unsafe cast between the two types when I need to.

Reply:
I made some fixes in svdtools, and svdtools 0.3.14 can now handle patching these registers to be derived.

I updated the svdtools version in Cargo.toml and Cargo.lock as part of this PR, but I can split that change out into a separate PR if you prefer.